src/util.js

```diff
@@ -68,11 +75,98 @@ function nintendoPasswordHash(password, pid) {
 	return hashed;
 }
 
+async function handleStripeEvent(event) {
+	if (event.type === 'customer.subscription.updated' || event.type === 'customer.subscription.deleted') {
+		const subscription = event.data.object;
+		const product = await stripe.products.retrieve(subscription.plan.product);
+		const customer = await stripe.customers.retrieve(subscription.customer);
+
+		if (!customer.metadata.pnid_pid && subscription.status !== 'canceled' && subscription.status !== 'unpaid') {
+			// No PNID PID linked to customer. Abort and refund!
+			await stripe.subscriptions.del(subscription.id);
+
+			const invoice = await stripe.invoices.retrieve(subscription.latest_invoice);
+			await stripe.refunds.create({
+				payment_intent: invoice.payment_intent
+			});
+
+			await sendGmail({
+				to: customer.email,
+				subject: 'Pretendo Subscription Failed - No Linked PNID',
+				text: `Your recent subscription to Pretendo Network has failed.\nThis is due to no PNID PID being linked to the Stripe customer account used. The subscription has been canceled and refunded. Please contact Jon immediately.\nStripe Customer ID: ${customer.id}`
+			});
+
+			return;
+		}
+
+		const pid = Number(customer.metadata.pnid_pid);
+		const pnid = await database.PNID.findOne({ pid });
+
+		const latestWebhookTimestamp = pnid.get('connections.stripe.latest_webhook_timestamp');
+
+		if (latestWebhookTimestamp && latestWebhookTimestamp > event.created) {
+			// Do nothing, this webhook is older than the latest seen
+			return;
+		}
+
+		if (!pnid && subscription.status !== 'canceled' && subscription.status !== 'unpaid') {
+			// PNID does not exist. Abort and refund!
+			await stripe.subscriptions.del(subscription.id);
+
+			const invoice = await stripe.invoices.retrieve(subscription.latest_invoice);
+			await stripe.refunds.create({
+				payment_intent: invoice.payment_intent
+			});
+
+			await sendGmail({
```

Comment:
Maybe do retry logic. If email is invalid this probably throws on email bounce. Making it infinite loop the events again.
(idk the library tho, idk if it throws on bounce)

(Email is user input, don't let it screw up the system)

So retry like 5 times otherwise give up. (The log from earlier should make sure we see it if it happens)